src/azure-cli/azure/cli/command_modules/backup/custom_afs.py

```diff
@@ -210,14 +210,23 @@ def restore_AzureFileShare(cmd, client, resource_group_name, vault_name, rp_name
 
 
 def list_recovery_points(cmd, client, resource_group_name, vault_name, item, start_date=None, end_date=None,
-                         use_secondary_region=None):
+                         use_secondary_region=None, is_ready_for_move=None, target_tier=None, tier=None,
+                         recommended_for_archive=None):
     if use_secondary_region:
         raise InvalidArgumentValueError(
             """
             --use-secondary-region flag is not supported for --backup-management-type AzureStorage.
             Please either remove the flag or query for any other backup-management-type.
             """)
 
+    if is_ready_for_move is not None or target_tier is not None or tier is not None:
+        raise InvalidArgumentValueError("""Invalid argument has been passed. --is-ready-for-move true, --target-tier
+        and --tier flags are not supported for --backup-management-type AzureStorage.""")
+
+    if recommended_for_archive is not None:
+        raise InvalidArgumentValueError("""--recommended-for-archive is supported by AzureIaasVM backup management
+        type only.""")
```

Comment:
`InvalidArgumentValueError` is for the case that the input value used for parameter is illegal. However, the problem here is that the parameters are not supported, so the error type `ArgumentUsageError` is more appropriate here, right?

Reply:
done